### PR TITLE
rgw: reduce default thread pool size

### PR DESCRIPTION
**Pull request by mkogan1**
Fixes: https://tracker.ceph.com/issues/65656

Attaching DFG Lab testing results:
image
image

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by cbodley — APPROVED** [review]
🚀

**Comment by mkogan1** [comment]
@cbodley in testing the memory footprint of 512 -vs- 128 threads encountered an issue when running 128 threads and performing 1000 concurrent 4KB GET requests with hsbench,
the MAX latency occasionally spikes on some of the 1 Milioon GET runs iterations:

`❯ time (nice numactl -N 1 -m 1 -- ~/go/bin/hsbench -a b2345678901234567890 -s b234567890123456789012345678901234567890 -u http://127.0.0.1:8000 -z 4K -d -1 -t 900 -b $(( $(numactl -N 0 -- nproc) * 2 )) -n 1000000 -m g -bp b01b- -op 'folder01/stage01_' `

```
# 128 threads: note the 'max: 79889.3' on the far right
2024/05/02 09:55:19 Loop: 0, Int: TOTAL, Dur(s): 112.6, Mode: GET, Ops: 999996, MB/s: 34.69, IO/s: 8880, Lat(ms): [ min: 0.6, avg: 36.5, 99%: 229.2, max: 79889.3 ]

# 512 threads: note the ' max: 2970.1' on the far right
2024/05/02 10:08:28 Loop: 0, Int: TOTAL, Dur(s): 38.7, Mode: GET, Ops: 1000000, MB/s: 100.89, IO/s: 25829, Lat(ms): [ min: 0.7, avg: 27.3, 99%: 221.8, max: 2970.1 ]

```

**Comment by cbodley** [comment]
@mkogan1 wow, 79 seconds is extreme, but the 99th percentile latency looks similar. can you find any evidence of slow osd ops that might have caused this? can we rule out other things like pool creations or pg splits that might cause delays like that?

**Comment by markhpc** [comment]
HDD+NVMe 12 OSD, RGW Efficiency (MB_s _ Core)
HDD+NVMe 12 OSD, RGW 4MB CPU Usage
HDD+NVMe 12 OSD, RGW 4MB Throughput (MB_s)

**Comment by mkogan1** [comment]
@markhpc nice, presume the _3xclients_ are hsbench, how many threads in `hsbench ... -t #` were specified during the test (interested from the perspective of how the RGWs handle the requests when `qactive`  is close to the `rgw_max_concurrent_requests=1024` (if it is even possible to get close to it))

**Comment by markhpc** [comment]
> @markhpc nice, presume the _3xclients_ are hsbench, how many threads in `hsbench ... -t #` were specified during the test (interested from the perspective of how the RGWs handle the requests when `qactive` is close to the `rgw_max_concurrent_requests=1024` (if it is even possible to get close to it))

I don't remember, I'll have to go back and check.  This was data from over a year ago.  Assume fairly high thread counts on the hsbench side though.

**Review by markhpc — APPROVED** [review]
Only concern is with the potential for introduction of high latency events, but otherwise I'm 100% for this change.

**Comment by markhpc** [comment]
We still want this

**Comment by markhpc** [comment]
@mkogan1 FYI, we've adopted this change for Clyso's downstream version of Ceph.  If we have major problems I'll report back, but we feel the testing is showing enough benefit that this is a better default for us at least.

**Comment by clwluvw** [comment]
Is this going to affect the number of CURL requests rgw can make? (asking for encryption case where rgw needs to call vault).

**Comment by cbodley** [comment]
> Is this going to affect the number of CURL requests rgw can make? (asking for encryption case where rgw needs to call vault and if I'm not mistaken those calls are async and are using that context pool(?)).

@clwluvw our curl client has supported coroutines since octopus with https://github.com/ceph/ceph/pull/25355, but the coroutine's yield context wasn't threaded into the kms requests (vault/barbican) until squid with https://github.com/ceph/ceph/pull/53684. before that, the vault requests would each tie up a thread so be limited by rgw_thread_pool_size

**Comment by markhpc** [comment]
> > Is this going to affect the number of CURL requests rgw can make? (asking for encryption case where rgw needs to call vault and if I'm not mistaken those calls are async and are using that context pool(?)).
> 
> @clwluvw our curl client has supported coroutines since octopus with #25355, but the coroutine's yield context wasn't threaded into the kms requests (vault/barbican) until squid with #53684. before that, the vault requests would each tie up a thread so be limited by rgw_thread_pool_size

That's good information.  We'll try to watch for that as our base image is still 17.2.7.

**Comment by Matt1360** [comment]
Hi folks, we've been kicking the `stale` can for a while on this one. It looks like a pretty clear win, and Clyso (and likely other folks) have adopted this and carried it for a while. Do we need to move this through QA?

**Comment by mkogan1** [comment]
> Hi folks, we've been kicking the `stale` can for a while on this one. It looks like a pretty clear win, and Clyso (and likely other folks) have adopted this and carried it for a while. Do we need to move this through QA?

Hi, yes, a long time has passed since the last test, and there were numerous changes to ceph/rgw aio -
re-tested on the current main:
```
❯ git branch -vv
* main fa93d68191b [origin/main] Merge pull request #65104 from edwinzrodriguez/ceph-wip-72556
```

TL;DR: Results are better but still ambiguous - 
workloads are PUT & GET 4KB objs using 900 threads:

1at PUT (`-m ip`):
```
❯ time (nice numactl -N 1 -m 1 -- ~/go/bin/hsbench -a b2345678901234567890 -s b234567890123456789012345678901234567890 -u http://127.0.0.1:8000 -z 4K -d -1 -t 900 -b $(( $(numactl -N 0 -- nproc) * 2 )) -n 1000000 -m ip -bp b01b- -op 'folder01/stage01_')
```
followed by GET (`-m g`):
```
❯ time (nice numactl -N 1 -m 1 -- ~/go/bin/hsbench -a b2345678901234567890 -s b234567890123456789012345678901234567890 -u http://127.0.0.1:8000 -z 4K -d -1 -t 900 -b $(( $(numactl -N 0 -- nproc) * 2 )) -n 1000000 -m g -bp b01b- -op 'folder01/stage01_')
```

1st for reference `rgw_thread_pool_size=512` -
PUT:
image
GET:
image

then `rgw_thread_pool_size=128` -
PUT:
image
GET:
image

The large 99 percentile latency spike seen in previous tests is no-more, which is nice. 
And the PUT workload shows clear perf advantage with 128 threads compared to 512 threads
But the GET workload shows slightly lower perf with 128 threads

so it's close, but still not a universal advantage to 128 threads, a user can choose 512 or 128 threads depending on his workload being read or write focused.


@cbodley @adamemerson, please advise if have any conf suggestions or unmerged PR to try, will happily test.

until next time...

**Comment by mkogan1** [comment]
re-tested on current main:
```
6ce249e 2025-12-07 10:16 +0200 Matan Breizman                            Merge pull request #66158 from xxhdx1985126/wip-seastore-minor-perf-issue
```

The latencies have improved since the previous benchmark!
(thou the GET OP/s are still lower with 128 threads compared to 512)


```
// reference rgw_thread_pool_size=512 -

PUT:
2025/12/07 13:23:59 Loop: 0, Int: TOTAL, Dur(s): 82.0, Mode: PUT, Ops: 1000000, MB/s: 47.64, IO/s: 12196, Lat(ms): [ min: 5.4, avg: 73.7, 99%: 82.8, max: 374.8 ], Slowdowns: 0 

GET:
2025/12/07 13:27:27 Loop: 0, Int: TOTAL, Dur(s): 37.2, Mode: GET, Ops: 1000000, MB/s: 105.05, IO/s: 26894, Lat(ms): [ min: 0.6, avg: 30.6, 99%: 119.5, max: 572.7 ], Slowdowns: 0


// rgw_thread_pool_size=128 -

PUT:
2025/12/07 13:32:03 Loop: 0, Int: TOTAL, Dur(s): 75.6, Mode: PUT, Ops: 1000000, MB/s: 51.67, IO/s: 13228, Lat(ms): [ min: 1.3, avg: 67.9, 99%: 76.3, max: 324.7 ], Slowdowns: 0

GET:
2025/12/07 13:34:31 Loop: 0, Int: TOTAL, Dur(s): 53.7, Mode: GET, Ops: 1000000, MB/s: 72.71, IO/s: 18615, Lat(ms): [ min: 0.6, avg: 37.6, 99%: 213.5, max: 888.2 ], Slowdowns: 0

```

**Comment by mkogan1** [comment]
@cbodley @markhpc @adamemerson @mattbenjamin 

Re-tested again on current _main_
```
git branch -vv
* main                    59a0bd8dd0b [origin/main: behind 7] Merge pull request #68026 from rhcs-dashboard/fix-theme
```

This time there is a performance **improvment** on all accountes --
`rgw_thread_pool_size=128` shows improvement in **both** higher _IO/s_ and lower _latency_ for **both** PUTs and GETs compared to `512` threads !!!
(which was not the case previously) 


Workload: PUT followed by GET of 1M 4KB objs using 900 concurrent requests:
(`rgw_max_concurrent_requests=1024`)

```
time (nice numactl -N 1 -m 1 -- ~/go/bin/hsbench -a b2345678901234567890 -s b234567890123456789012345678901234567890 -u http://127.0.0.1:8000 -z 4K -d -1 -t 900 -b $(( $(numactl -N 0 -- nproc) * 2 )) -n 1000000 -m ip -bp b01b- -op 'folder01/stage01_')

time (nice numactl -N 1 -m 1 -- ~/go/bin/hsbench -a b2345678901234567890 -s b234567890123456789012345678901234567890 -u http://127.0.0.1:8000 -z 4K -d -1 -t 900 -b $(( $(numactl -N 0 -- nproc) * 2 )) -n 1000000 -m g -bp b01b- -op 'folder01/stage01_')
```


Performance data collection from 3 machines:
(NOTE that in ALL the permutations the IO/s is higher and latency is lower)

1. High perf NVME:
```
❯ sudo nvme list
Node                  Generic               SN                   Model                                    Namespace  Usage                      Format           FW Rev  
--------------------- --------------------- -------------------- ---------------------------------------- ---------- -------------------------- ---------------- --------
/dev/nvme0n1          /dev/ng0n1            S5CVNA0N401866       Dell Express Flash PM1725b 1.6TB AIC     0x1          1.60  TB /   1.60  TB    512   B +  8 B   1.0.0   
```
vvv  scroll to the right:
```
> PUT: reference `rgw_thread_pool_size=512`
2026/04/27 14:39:31 Loop: 0, Int: TOTAL, Dur(s): 91.7, Mode: PUT, Ops: 1000000, MB/s: 42.61, IO/s: 10909, Lat(ms): [ min: 2.3, avg: 82.4, 99%: 93.2, max: 190.7 ], Slowdowns: 0
> PUT: `rgw_thread_pool_size=128`
2026/04/27 14:44:36 Loop: 0, Int: TOTAL, Dur(s): 83.5, Mode: PUT, Ops: 1000000, MB/s: 46.79, IO/s: 11979, Lat(ms): [ min: 4.4, avg: 75.1, 99%: 83.6, max: 161.2 ], Slowdowns: 0
...................................................................................................^^^^^.............................................^^^^^^^^^^

> GET: reference `rgw_thread_pool_size=512`
2026/04/27 14:40:16 Loop: 0, Int: TOTAL, Dur(s): 36.2, Mode: GET, Ops: 1000000, MB/s: 107.78, IO/s: 27592, Lat(ms): [ min: 0.8, avg: 32.5, 99%: 96.2, max: 617.1 ], Slowdowns: 0
> GET: `rgw_thread_pool_size=128`
2026/04/27 14:45:19 Loop: 0, Int: TOTAL, Dur(s): 30.6, Mode: GET, Ops: 1000000, MB/s: 127.54, IO/s: 32651, Lat(ms): [ min: 0.7, avg: 26.8, 99%: 74.9, max: 281.0 ], Slowdowns: 0
...................................................................................................^^^^^.............................................^^^^^^^^^^
```

2. Medium perf NVME:
```
❯ sudo nvme list                                                                                                                                                                                  
Node                  Generic               SN                   Model                                    Namespace  Usage                      Format           FW Rev                           
--------------------- --------------------- -------------------- ---------------------------------------- ---------- -------------------------- ---------------- --------
/dev/nvme0n1          /dev/ng0n1            24464CFE3346         Micron_7450_MTFDKCC1T6TFS                0x1          1.03  TB /   1.60  TB    512   B +  0 B   E2MU200 
```
vvv  scroll to the right:
```
> PUT: reference `rgw_thread_pool_size=512`
2026/04/28 07:16:29 Loop: 0, Int: TOTAL, Dur(s): 139.4, Mode: PUT, Ops: 1000000, MB/s: 28.02, IO/s: 7173, Lat(ms): [ min: 21.9, avg: 125.4, 99%: 152.2, max: 440.6 ], Slowdowns: 0
> PUT: `rgw_thread_pool_size=128`
2026/04/28 07:26:05 Loop: 0, Int: TOTAL, Dur(s): 137.0, Mode: PUT, Ops: 1000000, MB/s: 28.52, IO/s: 7300, Lat(ms): [ min: 24.0, avg: 123.2, 99%: 151.1, max: 272.1 ], Slowdowns: 0
...................................................................................................^^^^^.............................................^^^^^^^^^^

> GET: reference `rgw_thread_pool_size=512`
2026/04/28 07:18:48 Loop: 0, Int: TOTAL, Dur(s): 61.1, Mode: GET, Ops: 1000000, MB/s: 63.89, IO/s: 16355, Lat(ms): [ min: 0.8, avg: 54.8, 99%: 173.5, max: 223.3 ], Slowdowns: 0
> GET: `rgw_thread_pool_size=128`
2026/04/28 07:30:02 Loop: 0, Int: TOTAL, Dur(s): 58.4, Mode: GET, Ops: 1000000, MB/s: 66.85, IO/s: 17114, Lat(ms): [ min: 0.6, avg: 52.4, 99%: 168.8, max: 228.9 ], Slowdowns: 0
...................................................................................................^^^^^.............................................^^^^^^^^^^

```

3. HDD:
```
❯ cat /proc/scsi/scsi
Attached devices:
Host: scsi11 Channel: 02 Id: 00 Lun: 00
  Vendor: AVAGO    Model: SMC3108          Rev: 4.29
  Type:   Direct-Access                    ANSI  SCSI revision: 05
```
vvv  scroll to the right:
```
> PUT: reference `rgw_thread_pool_size=512`
2026/04/28 14:12:10 Loop: 0, Int: TOTAL, Dur(s): 223.8, Mode: PUT, Ops: 1000000, MB/s: 17.45, IO/s: 4467, Lat(ms): [ min: 1.8, avg: 201.4, 99%: 233.1, max: 2576.0 ], Slowdowns: 0
> PUT: `rgw_thread_pool_size=128`
2026/04/28 14:26:06 Loop: 0, Int: TOTAL, Dur(s): 136.4, Mode: PUT, Ops: 1000000, MB/s: 28.64, IO/s: 7332, Lat(ms): [ min: 4.9, avg: 122.7, 99%: 158.9, max: 314.1 ], Slowdowns: 0
...................................................................................................^^^^^.............................................^^^^^^^^^^

> GET: reference `rgw_thread_pool_size=512`
2026/04/28 14:17:26 Loop: 0, Int: TOTAL, Dur(s): 272.1, Mode: GET, Ops: 1000000, MB/s: 14.35, IO/s: 3675, Lat(ms): [ min: 12.4, avg: 244.8, 99%: 678.6, max: 1319.9 ], Slowdowns: 0
> GET: `rgw_thread_pool_size=128`
2026/04/28 14:30:48 Loop: 0, Int: TOTAL, Dur(s): 227.3, Mode: GET, Ops: 1000000, MB/s: 17.18, IO/s: 4399, Lat(ms): [ min: 8.8, avg: 204.4, 99%: 393.4, max: 678.0 ], Slowdowns: 0
...................................................................................................^^^^^.............................................^^^^^^^^^^

```

**Comment by mkogan1** [comment]
jenkins retest this please

**Comment by mkogan1** [comment]
jenkins test docs

**Comment by mkogan1** [comment]
jenkins render docs

**Comment by markhpc** [comment]
> @cbodley @markhpc @adamemerson @mattbenjamin
> 
> Re-tested again on current _main_
> 
> ```
> git branch -vv
> * main                    59a0bd8dd0b [origin/main: behind 7] Merge pull request #68026 from rhcs-dashboard/fix-theme
> ```
> 
> This time there is a performance **improvment** on all accountes -- `rgw_thread_pool_size=128` shows improvement in **both** higher _IO/s_ and lower _latency_ for **both** PUTs and GETs compared to `512` threads !!! (which was not the case previously)
> 
> Workload: PUT followed by GET of 1M 4KB objs using 900 concurrent requests: (`rgw_max_concurrent_requests=1024`)
> 
> ```
> time (nice numactl -N 1 -m 1 -- ~/go/bin/hsbench -a b2345678901234567890 -s b234567890123456789012345678901234567890 -u http://127.0.0.1:8000 -z 4K -d -1 -t 900 -b $(( $(numactl -N 0 -- nproc) * 2 )) -n 1000000 -m ip -bp b01b- -op 'folder01/stage01_')
> 
> time (nice numactl -N 1 -m 1 -- ~/go/bin/hsbench -a b2345678901234567890 -s b234567890123456789012345678901234567890 -u http://127.0.0.1:8000 -z 4K -d -1 -t 900 -b $(( $(numactl -N 0 -- nproc) * 2 )) -n 1000000 -m g -bp b01b- -op 'folder01/stage01_')
> ```
> 
> Performance data collection from 3 machines: (NOTE that in ALL the permutations the IO/s is higher and latency is lower)
> 
> 1. High perf NVME:
> 
> ```
> ❯ sudo nvme list
> Node                  Generic               SN                   Model                                    Namespace  Usage                      Format           FW Rev  
> --------------------- --------------------- -------------------- ---------------------------------------- ---------- -------------------------- ---------------- --------
> /dev/nvme0n1          /dev/ng0n1            S5CVNA0N401866       Dell Express Flash PM1725b 1.6TB AIC     0x1          1.60  TB /   1.60  TB    512   B +  8 B   1.0.0   
> ```
> 
> vvv scroll to the right:
> 
> ```
> > PUT: reference `rgw_thread_pool_size=512`
> 2026/04/27 14:39:31 Loop: 0, Int: TOTAL, Dur(s): 91.7, Mode: PUT, Ops: 1000000, MB/s: 42.61, IO/s: 10909, Lat(ms): [ min: 2.3, avg: 82.4, 99%: 93.2, max: 190.7 ], Slowdowns: 0
> > PUT: `rgw_thread_pool_size=128`
> 2026/04/27 14:44:36 Loop: 0, Int: TOTAL, Dur(s): 83.5, Mode: PUT, Ops: 1000000, MB/s: 46.79, IO/s: 11979, Lat(ms): [ min: 4.4, avg: 75.1, 99%: 83.6, max: 161.2 ], Slowdowns: 0
> ...................................................................................................^^^^^.............................................^^^^^^^^^^
> 
> > GET: reference `rgw_thread_pool_size=512`
> 2026/04/27 14:40:16 Loop: 0, Int: TOTAL, Dur(s): 36.2, Mode: GET, Ops: 1000000, MB/s: 107.78, IO/s: 27592, Lat(ms): [ min: 0.8, avg: 32.5, 99%: 96.2, max: 617.1 ], Slowdowns: 0
> > GET: `rgw_thread_pool_size=128`
> 2026/04/27 14:45:19 Loop: 0, Int: TOTAL, Dur(s): 30.6, Mode: GET, Ops: 1000000, MB/s: 127.54, IO/s: 32651, Lat(ms): [ min: 0.7, avg: 26.8, 99%: 74.9, max: 281.0 ], Slowdowns: 0
> ...................................................................................................^^^^^.............................................^^^^^^^^^^
> ```
> 
> 2. Medium perf NVME:
> 
> ```
> ❯ sudo nvme list                                                                                                                                                                                  
> Node                  Generic               SN                   Model                                    Namespace  Usage                      Format           FW Rev                           
> --------------------- --------------------- -------------------- ---------------------------------------- ---------- -------------------------- ---------------- --------
> /dev/nvme0n1          /dev/ng0n1            24464CFE3346         Micron_7450_MTFDKCC1T6TFS                0x1          1.03  TB /   1.60  TB    512   B +  0 B   E2MU200 
> ```
> 
> vvv scroll to the right:
> 
> ```
> > PUT: reference `rgw_thread_pool_size=512`
> 2026/04/28 07:16:29 Loop: 0, Int: TOTAL, Dur(s): 139.4, Mode: PUT, Ops: 1000000, MB/s: 28.02, IO/s: 7173, Lat(ms): [ min: 21.9, avg: 125.4, 99%: 152.2, max: 440.6 ], Slowdowns: 0
> > PUT: `rgw_thread_pool_size=128`
> 2026/04/28 07:26:05 Loop: 0, Int: TOTAL, Dur(s): 137.0, Mode: PUT, Ops: 1000000, MB/s: 28.52, IO/s: 7300, Lat(ms): [ min: 24.0, avg: 123.2, 99%: 151.1, max: 272.1 ], Slowdowns: 0
> ...................................................................................................^^^^^.............................................^^^^^^^^^^
> 
> > GET: reference `rgw_thread_pool_size=512`
> 2026/04/28 07:18:48 Loop: 0, Int: TOTAL, Dur(s): 61.1, Mode: GET, Ops: 1000000, MB/s: 63.89, IO/s: 16355, Lat(ms): [ min: 0.8, avg: 54.8, 99%: 173.5, max: 223.3 ], Slowdowns: 0
> > GET: `rgw_thread_pool_size=128`
> 2026/04/28 07:30:02 Loop: 0, Int: TOTAL, Dur(s): 58.4, Mode: GET, Ops: 1000000, MB/s: 66.85, IO/s: 17114, Lat(ms): [ min: 0.6, avg: 52.4, 99%: 168.8, max: 228.9 ], Slowdowns: 0
> ...................................................................................................^^^^^.............................................^^^^^^^^^^
> ```
> 
> 3. HDD:
> 
> ```
> ❯ cat /proc/scsi/scsi
> Attached devices:
> Host: scsi11 Channel: 02 Id: 00 Lun: 00
>   Vendor: AVAGO    Model: SMC3108          Rev: 4.29
>   Type:   Direct-Access                    ANSI  SCSI revision: 05
> ```
> 
> vvv scroll to the right:
> 
> ```
> > PUT: reference `rgw_thread_pool_size=512`
> 2026/04/28 14:12:10 Loop: 0, Int: TOTAL, Dur(s): 223.8, Mode: PUT, Ops: 1000000, MB/s: 17.45, IO/s: 4467, Lat(ms): [ min: 1.8, avg: 201.4, 99%: 233.1, max: 2576.0 ], Slowdowns: 0
> > PUT: `rgw_thread_pool_size=128`
> 2026/04/28 14:26:06 Loop: 0, Int: TOTAL, Dur(s): 136.4, Mode: PUT, Ops: 1000000, MB/s: 28.64, IO/s: 7332, Lat(ms): [ min: 4.9, avg: 122.7, 99%: 158.9, max: 314.1 ], Slowdowns: 0
> ...................................................................................................^^^^^.............................................^^^^^^^^^^
> 
> > GET: reference `rgw_thread_pool_size=512`
> 2026/04/28 14:17:26 Loop: 0, Int: TOTAL, Dur(s): 272.1, Mode: GET, Ops: 1000000, MB/s: 14.35, IO/s: 3675, Lat(ms): [ min: 12.4, avg: 244.8, 99%: 678.6, max: 1319.9 ], Slowdowns: 0
> > GET: `rgw_thread_pool_size=128`
> 2026/04/28 14:30:48 Loop: 0, Int: TOTAL, Dur(s): 227.3, Mode: GET, Ops: 1000000, MB/s: 17.18, IO/s: 4399, Lat(ms): [ min: 8.8, avg: 204.4, 99%: 393.4, max: 678.0 ], Slowdowns: 0
> ...................................................................................................^^^^^.............................................^^^^^^^^^^
> ```

That's fantastic.  If you are feeling confident about merging this now, I'm in favor as well.  Should we remove the DNM tag?

**Comment by adamemerson** [comment]
I'd very much like to move this forward if @ivancich doesn't object.

**Comment by mkogan1** [comment]
submitted a teuthology run with this commit rebased over the current main:
`4bee572 2026-05-10 11:03 +0300 Matan Breizman               M─┐ Merge pull request #68750 from tchaikov/wip-crimson-merge-coll`

at:
https://pulpito-ng.ceph.com/runs/mkogan-2026-05-10_13:20:50-rgw-wip-rgw-thread-pool-size_i01-distro-default-trial

**Comment by cbodley** [comment]
> at: https://pulpito-ng.ceph.com/runs/mkogan-2026-05-10_13:20:50-rgw-wip-rgw-thread-pool-size_i01-distro-default-trial

thank you @mkogan1, that looks like a pass :+1:

this is probably worth mentioning in the release notes for umbrella. could you please add some text to PendingReleaseNotes?

**Comment by mkogan1** [comment]
@cbodley , re:
> this is probably worth mentioning in the release notes for umbrella. could you please add some text to PendingReleaseNotes?

Done, PTAL if in correct place and wording please.

**Review by cbodley — APPROVED** [review]
thanks for the release note!

**Comment by cbodley** [comment]
jenkins retest this please
